xl_auth/commands.py

```diff
@@ -172,3 +172,254 @@ def urls(url, order):
+            bibdb_cataloging_admin_emails_and_names[cataloging_admin_email] = cataloging_admin_name
+            assert bibdb_sigel != ''
+            registering_bibdb_sigels.add(bibdb_sigel)
+            if not cataloging_admin_email:
+                continue
+
+            bibdb_cataloging_admins.add(cataloging_admin_email)
+            bibdb_sigels_and_cataloging_admins[bibdb_sigel] = cataloging_admin_email
+
+            if cataloging_admin_email in bibdb_cataloging_admins_and_sigels:
+                bibdb_cataloging_admins_and_sigels[cataloging_admin_email].add(bibdb_sigel)
+            else:
+                bibdb_cataloging_admins_and_sigels[cataloging_admin_email] = {bibdb_sigel}
+
+        print('registering_bibdb_sigels:', len(registering_bibdb_sigels), '/',
+              'bibdb_cataloging_admins:', len(bibdb_cataloging_admins))
+
+        return {
+            'sigel_to_cataloging_admins': bibdb_sigels_and_cataloging_admins,
+            'cataloging_admin_to_sigels': bibdb_cataloging_admins_and_sigels,
+            'cataloging_admin_emails_to_names': bibdb_cataloging_admin_emails_and_names,
+            'sigels': registering_bibdb_sigels,
+            'cataloging_admins': bibdb_cataloging_admins,
+        }
+
+    def _get_bibdb_sigels_not_in_voyager(bibdb_sigels, voyager_sigels):
+        unknown_sigels = set()
+        for bibdb_sigel in bibdb_sigels:
+            if bibdb_sigel not in voyager_sigels:
+                unknown_sigels.add(bibdb_sigel)
+        return unknown_sigels
+
+    def _generate_xl_auth_cataloging_admins_and_collections(bibdb_cataloging_admin_to_sigels,
+                                                            voyager_sigel_to_collections):
+        pre_total, post_total = 0, 0
+        voyager_sigels_unknown_in_bibdb = set()
+        xl_auth_cataloging_admins = dict()
+        xl_auth_collections = dict()
+        for cataloging_admin, sigels in bibdb_cataloging_admin_to_sigels.items():
+            pre_total += len(sigels)
+            xl_auth_cataloging_admins[cataloging_admin] = set()
+            for sigel in sigels:
+                if sigel not in xl_auth_collections:
+                    xl_auth_collections[sigel] = _get_collection_details_from_bibdb(sigel)
+
+                xl_auth_cataloging_admins[cataloging_admin].add(sigel)
+                if sigel in bibdb_sigels_unknown_in_voyager:
+                    continue
+
+                for voyager_collection in voyager_sigel_to_collections[sigel]:
+                    if voyager_collection not in xl_auth_collections:
+                        try:
+                            xl_auth_collections[voyager_collection] = \
+                                _get_collection_details_from_bibdb(voyager_collection)
+                        except AssertionError:
+                            voyager_sigels_unknown_in_bibdb.add(voyager_collection)
+                            continue  # TODO: Review me!
```

Comment:
I don't understand the TODO. Who is it for and that is `me` in this context?

Reply:
`me` would be the *continue* statement. 😄 Is it needed or not? I think not... Removing it.

Reply:
Interesting thing.. When replying to this comment on *Files Changed* tab, the comment does not show up here. Your comments are in both..

Reply:
`me` would be the *continue* statement. 😄 Is it needed or not? I think not... Removing it.

Reply:
Correction: It IS needed.